src/main/java/com/google/api/codegen/transformer/OutputTransformer.java

```diff
@@ -207,8 +200,48 @@ public void run() {
     TypeModel type;
     if (baseIdentifier.equals(RESPONSE_PLACEHOLDER)) {
       view.variable(context.getNamer().getSampleResponseVarName(context));
+      boolean pageStreaming = context.getMethodConfig().getPageStreaming() != null;
+      boolean pageStreamingAndUseResourceName =
+          pageStreaming
+              ? context
+                  .getFeatureConfig()
+                  .useResourceNameFormatOption(
+                      context.getMethodConfig().getPageStreaming().getResourcesFieldConfig())
+              : false;
+
+      // Compute the resource name format of output type and save that to local variables
+      if (pageStreamingAndUseResourceName) {
+        String typeName =
+            context
+                .getNamer()
+                .getAndSaveElementResourceTypeName(
+                    context.getTypeTable(),
+                    context.getMethodConfig().getPageStreaming().getResourcesFieldConfig());
+
+        Preconditions.checkArgument(
+            config.scan() == Scanner.EOF,
+            "%s:%s: accessing fields of resource names is not supported",
+            context.getMethodModel().getSimpleName(),
+            valueSet.getId());
 
-      if (context.getMethodConfig().getPageStreaming() != null) {
+        if (newVar != null) {
+          Preconditions.checkArgument(
+              !intoScalar,
+              "%s:%s: cannot parse a resource name into scalar",
+              context.getMethodModel().getSimpleName(),
+              valueSet.getId());
+
+          if (!localVars.put(newVar, null, typeName)) {
+            throw new IllegalStateException(
```

Comment:
This should be `IllegalArgumentException` I think, because user gave us bad config. You can also use `checkArgument` here.